vassal-app/src/main/java/VASSAL/build/module/PrototypesContainer.java

```diff
@@ -95,26 +95,19 @@ public static String getConfigureTypeName() {
     return Resources.getString("Editor.PrototypesContainer.component_type"); //$NON-NLS-1$
   }
 
-
-  private void rebuildPrototypeMap(AbstractBuildable target) {
-    for (final Buildable b : target.getBuildables()) {
-      if (b instanceof PrototypeDefinition) {
-        addDefinition((PrototypeDefinition)b);
-      }
-      else if (b instanceof AbstractBuildable) {
-        rebuildPrototypeMap((AbstractBuildable)b);
-      }
-    }
+  // Rebuild prototype cross-reference
+  private void rebuildPrototypeMap() {
+    definitions.clear();
+    GameModule.getGameModule().getAllDescendantComponentsOf(PrototypeDefinition.class).forEach((def) -> definitions.put(def.getConfigureName(), def));
   }
 
-
   public void addDefinition(PrototypeDefinition def) {
     definitions.put(def.getConfigureName(), def);
     def.addPropertyChangeListener(evt -> {
       if (NAME_PROPERTY.equals(evt.getPropertyName())) {
         // When a prototype is renamed we need to rebuild the prototype map, so that if there was a duplicate of the same name it will re-establish its presence
         definitions.clear();
```

Comment:
Isn't this line now redundant (`definitions.clear()`) is now the first command of `rebuildPrototypeMap()`)?

Reply:
Indeed. Thanks. Update pushed.